test/new-e2e/utils/e2e/client/agent.go

```diff
@@ -30,6 +35,56 @@ func (agent *Agent) init(auth *Authentification, data *agent.ClientData) error {
 	return err
 }
 
-func (agent *Agent) Status() (string, error) {
-	return agent.sshClient.Execute("sudo datadog-agent status")
+func (agent *Agent) Version() (string, error) {
+	return agent.sshClient.Execute("datadog-agent version")
+}
+
+type Status struct {
+	rawString string
+}
+
+func NewStatus(s string) *Status {
+	return &Status{rawString: s}
+}
+
+func (agent *Agent) Status() (*Status, error) {
+	s, err := agent.sshClient.Execute("sudo datadog-agent status")
+	if err != nil {
+		return nil, err
+	}
+	return NewStatus(s), nil
+}
+
+// IsReady true if status contains a valid version
+func (s *Status) IsReady() (bool, error) {
+	return regexp.MatchString("={15}\nAgent \\(v7\\.\\d{2}\\..*\n={15}", s.rawString)
+}
+
+// IsReady runs status command and returns true if the status is ready
+// Use this to wait for agent to be ready before running any command
+func (a *Agent) IsReady() (bool, error) {
```

Comment:
Do you see any use cases of this function as user will probably use `WaitForReady` instead?

Reply:
I can see use of this instead of the `WaitForReady` wrapping it explicitly with a backoff retrier. Another one is testing that the agent is not ready if a config is badly formatted.